src/main/java/brandkon/product/ProductService.java

```diff
@@ -0,0 +1,95 @@
+package brandkon.product;
+
+import brandkon.brand.Brand;
+import brandkon.brand.BrandRepository;
+import brandkon.category.Category;
+import brandkon.category.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ProductService {
+    ProductRepository productRepository;
+    BrandRepository brandRepository;
+    CategoryRepository categoryRepository;
+
+    public ProductService(ProductRepository productRepository, BrandRepository brandRepository, CategoryRepository categoryRepository) {
+        this.productRepository = productRepository;
+        this.brandRepository = brandRepository;
+        this.categoryRepository = categoryRepository;
+    }
+
+    List<ProductResponse> showAll(){
+       return productRepository.findAll()
+               .stream()
+               .map(product -> new ProductResponse(
+                       product.getId(),
+                       product.getBrand().getName(),
+                       product.getProductName(),
+                       product.getPrice(),
+                       product.getImageUrl()
+               )).toList();
+    }
+    ProductDetailResponse showId(Long id){
+
+        Product product = productRepository.findById(id).orElseThrow();
+        Brand brand = brandRepository.findById(product.getBrand().getId()).orElseThrow();
+        return new ProductDetailResponse(product.getId(),
+                product.getBrand().getName(),
+                product.getPrice(),
+                new BrandDetailResponses(brand.getId(), brand.getName(), brand.getGuideLines()),
+                product.getExpirationDays());
+    }
+    List<ProductResponse> showCategoryPopular(Long id){
```

Comment:
```suggestion
    List<ProductResponse> showCategoryPopular(Long categoryId){
```
파라미터 이름을 더 상세하게 적는 것이 좋을 것 같아요. 그냥 `id`라고 하면 `ProductService`라서 `productId`라고 착각할 수 있을 것 같네요.